src/structarray.jl

```diff
@@ -463,9 +479,20 @@ Base.@pure cst(::Type{SA}) where {SA} = combine_style_types(array_types(SA).para
 
 BroadcastStyle(::Type{SA}) where {SA<:StructArray} = StructArrayStyle{typeof(cst(SA)), ndims(SA)}()
 
-function Base.similar(bc::Broadcasted{StructArrayStyle{S, N}}, ::Type{ElType}) where {S<:DefaultArrayStyle, N, ElType}
-    ContainerType = isnonemptystructtype(ElType) ? StructArray{ElType} : Array{ElType}
-    return similar(ContainerType, axes(bc))
+# Here we use `similar` defined for `S` to build the dest Array.
+function Base.similar(bc::Broadcasted{StructArrayStyle{S, N}}, ::Type{ElType}) where {S, N, ElType}
+    bc′ = convert(Broadcasted{S}, bc)
+    isnonemptystructtype(ElType) || return similar(bc′, ElType)
+    return buildfromschema(T -> similar(bc′, T), ElType)
```

Comment:
```suggestion
    return isnonemptystructtype(ElType) ? similar(bc′, ElType) : buildfromschema(T -> similar(bc′, T), ElType)
```

Same above, rather than

```julia
if cond
    return x1
end
return x2
```

I think `if .. else ...` is a bit easier to read (even though of course the `else` is not necessary because of `return`).